apps/storybook/src/Stacking.stories.mdx

```diff
@@ -0,0 +1,125 @@
+
+1. axis system (including the axis grid);
+2. SVG elements rendered via [`SvgElement`](http://localhost:6006/?path=/story/building-blocks-svgelement--default) into the `svg` overlay, in DOM order;
+3. HTML elements rendered via [`Html`](http://localhost:6006/?path=/story/building-blocks-html--default), in DOM order (including [`Annotation`](http://localhost:6006/?path=/story/building-blocks-annotation--default) and [`Overlay`](http://localhost:6006/?path=/story/building-blocks-overlay--default));
+4. tooltip
+5. floating toolbar (typically with reset zoom button)
+
+In most cases, within groups 2 and 3, **all you need to do to move an element `A` behind, or in front of, another element `B` is to render `A`
+respectively before or after `B` in your React tree**.
+
+However, there are some pitfalls:
+
+- Elements from groups 2 and 3 cannot be interleaved with one another. Consider the `svg` overlay as its own stacking context:
+  its children are stacked only in relation with one another within the SVG document.
+- When `Html` and `SvgElement` unmount and then remount (e.g. based on conditions), their children can end up moving down the DOM tree and therefore
+  in front of other elements that remained in the DOM. This can be worked around in a few ways, as explained below.
+
+### How to avoid unmounting `Html` and `SvgElement`
+
+This is also better for performance, as mounting `Html` and `SvgElement` is costly. The simplest solution is to move the rendering condition to the children:
+
+```tsx
+<>
+  {/* Avoid */}
+  {bool && (
+    <Annotation x={0} y={0}>
+      Foo
+    </Annotation>
+  )}
+  {bool && (
+    <Html>
+      <p>Bar</p>
+    </Html>
+  )}
+  {bool && (
+    <SvgElement>
+      <SvgRect />
+    </SvgElement>
+  )}
+
+  {/* Prefer */}
+  <Annotation x={0} y={0}>
+    {bool && 'Foo'}
+  </Annotation>
+  <Html>{bool && <p>Bar</p>}</Html>
+  <SvgElement>{bool && <SvgRect />}</SvgElement>
+</>
+```
+
+Another solution is to conditionally hide the children:
+
+```tsx
+<>
+  <Html x={0} y={0}>
+    <span hidden={bool || undefined}>Foo</span>
+  </Html>
+
+  {/* Hide `Annotation`'s internal `div` (works with `Overlay` as well) */}
+  <Annotation x={0} y={0} hidden={bool || undefined}>
+    Foo
+  </Annotation>
+
+  <SvgElement>
+    <SvgRect fill={bool ? 'skyblue' : 'transparent'} />
+  </SvgElement>
+</>
+```
+
+### What if DOM order of HTML elements cannot be changed or guaranteed?
+
+If you can't restructure your React tree to render `Html` elements in the order you want them to be stacked,
+or if you can't avoid unmounting/remounting them, the last resort is to control the stacking order explicitly with `z-index`.
+
+Knowing that, by default, `Html`'s internal `div` element has no `z-index`,
+which is equivalent to `z-index: 0`, you can move an `Html` element in front of,
+or behind, other `Html` elements simply by giving it a positive or negative
+`z-index` (respectively).
```

Comment:
Lots of info in this: I suggest to break it down in two sentences.

```suggestion
By default, `Html`'s internal `div` element has no `z-index`,
this is equivalent to `z-index: 0`. 
You can therefore move an `Html` element in front of,
or behind, other `Html` elements simply by giving it a positive or negative
`z-index` (respectively).
```

Reply:
Good call 👍